include/bluetooth/gatt.h

```diff
@@ -1217,10 +1217,12 @@ struct bt_gatt_exchange_params {
  *  @note Shall only be used once per connection.
  *
  *  The Response comes in callback @p params->func. The callback is run from
- *  the BT RX thread. @p params must remain valid until start of callback.
+ *  the BT RX thread or System work queue when @kconfig{CONFIG_BT_RECV_IS_RX_THREAD}
+ *  not selected. @p params must remain valid until start of callback.
  *
  *  This function will block while the ATT request queue is full, except when
- *  called from the BT RX thread, as this would cause a deadlock.
+ *  called from the BT RX thread or System work queue thread when
+ *  @kconfig{CONFIG_BT_RECV_IS_RX_THREAD} not selected, as this would cause a deadlock.
```

Comment:
I wonder if we could make it something simpler like 

```suggestion
 *  called from a Bluetooth callback, as this would cause a deadlock.
```

and still keep the same amount of information?